### PR TITLE
Pteron36, open source 3x5_3 supporting layout

### PR DESCRIPTION
**Pull request by harshitgoel96**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/pteron36/info.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    "keyboard_name": "Pteron36",
+    "url": "",
+    "maintainer": "Harshit Goel",
+    "width": 5,
+    "height": 8,
+    "layouts": {
+        "LAYOUT_split_3x5_3": {
+            "layout":  [{"x":2, "y":0}, {"x":12.5, "y":0}, {"x":1, "y":0.5}, {"x":3, "y":0.5}, {"x":11.5, "y":0.5}, {"x":13.5, "y":0.5}, {"x":4, "y":0.75}, {"x":10.5, "y":0.75}, {"x":0, "y":1}, {"x":2, "y":1}, {"x":12.5, "y":1}, {"x":14.5, "y":1}, {"x":1, "y":1.5}, {"x":3, "y":1.5}, {"x":11.5, "y":1.5}, {"x":13.5, "y":1.5}, {"x":4, "y":1.75}, {"x":10.5, "y":1.75}, {"x":0, "y":2}, {"x":2, "y":2}, {"x":12.5, "y":2}, {"x":14.5, "y":2}, {"x":1, "y":2.5}, {"x":3, "y":2.5}, {"x":11.5, "y":2.5}, {"x":13.5, "y":2.5}, {"x":4, "y":2.75}, {"x":10.5, "y":2.75}, {"x":0, "y":3}, {"x":14.5, "y":3}, {"x":4, "y":4}, {"x":5, "y":4}, {"x":6, "y":4}, {"x":8.5, "y":4}, {"x":9.5, "y":4}, {"x":10.5, "y":4}]
+        },
+        "LAYOUT_split_3x5_3_RotEnc": {
```
Comment (fauxpark) [inline keyboards/pteron36/info.json:11]:
```suggestion
        "LAYOUT_split_3x5_3_encoder": {
```

File: keyboards/pteron36/pteron36.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+     {L00, L01, L02,L03, L04 },\
+     {L10, L11, L12,L13, L14 },\
+    {L20, L21, L22,L23, L24 },\
+    {L30, L31, L32, KC_NO, KC_NO },\
+    {R00, R01, R02,R03, R04 },\
+    {R10, R11, R12,R13, R14 },\
+    {R20, R21, R22,R23, R24 },\
+    {R30, R31, R32,KC_NO, KC_NO }\
+}
+#define LAYOUT_split_3x5_3_rotenc(\
```
Comment (fauxpark) [inline keyboards/pteron36/pteron36.h:45]:
```suggestion
#define LAYOUT_split_3x5_3_encoder(\
```

File: keyboards/pteron36/pteron36.h [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+     {L00, L01, L02,L03, L04 },\
+     {L10, L11, L12,L13, L14 },\
+    {L20, L21, L22,L23, L24 },\
+    {L30, L31, L32, KC_NO, L34 },\
+    {R00, R01, R02,R03, R04 },\
+    {R10, R11, R12,R13, R14 },\
+    {R20, R21, R22,R23, R24 },\
+    {R30, R31, R32,KC_NO, R34 }\
+}
+//#define LAYOUT LAYOUT_split_3x5_3
```
Comment (fauxpark) [inline keyboards/pteron36/pteron36.h:60]:
```suggestion
```

File: keyboards/pteron36/readme.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Pteron36
+
+![Pteron36](https://github.com/harshitgoel96/pteron36-split-keyboard)
+
+*A short description of the keyboard/project*
```
Comment (fauxpark) [inline keyboards/pteron36/readme.md:5]:
```suggestion
```

File: keyboards/pteron36/readme.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+Flashing example for this keyboard:
+
+    make pteron36:default_split_3x5_3:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Default Keymap and Miryoku Layout support
+
+There is no default keymap for the keyboard as it supports community layout split_3x5_3, you can define your own layout in traditional way.
```
Comment (fauxpark) [inline keyboards/pteron36/readme.md:23]: All keyboards must have a default keymap.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]
.

**Comment by harshitgoel96** [comment]
info.json fixed

**Review by fauxpark — APPROVED** [review]

File: keyboards/pteron36/rules.mk [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+
+OLED_DRIVER_ENABLE = no    # OLED display; work in progress to add support. will be update in future.
+SPLIT_KEYBOARD = yes
+
+LAYOUTS = split_3x5_3
```
Comment (drashna) [inline keyboards/pteron36/rules.mk:27]:
```suggestion
LAYOUTS = split_3x5_3

```

Comment (harshitgoel96) [reply keyboards/pteron36/rules.mk:27]: just need to add new line after the layouts? not sure the exact suggested change

Comment (fauxpark) [reply keyboards/pteron36/rules.mk:27]: image

**Comment by harshitgoel96** [comment]
i have added support for via, can i add that into this pr as well?

**Comment by harshitgoel96** [comment]
support for via added in the same pr

File: keyboards/pteron36/readme.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
+# Pteron36
+
+![Pteron36](https://github.com/harshitgoel96/pteron36-split-keyboard)
```
Comment (fauxpark) [inline keyboards/pteron36/readme.md:3]: This URL isn't an image.

File: keyboards/pteron36/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+  */ 
+#include QMK_KEYBOARD_H
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+enum custom_keycodes {
```
Comment (fauxpark) [inline keyboards/pteron36/keymaps/via/keymap.c:24]:
```suggestion
enum layer_names {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/pteron36/readme.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
+# Pteron36
+
+![Pteron36](https://i.imgur.com/PuXPWYf.jpg)
```
Comment (fauxpark) [inline keyboards/pteron36/readme.md:3]:
```suggestion
![Pteron36](https://i.imgur.com/PuXPWYfh.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**